src/sentry/db/models/base.py

```diff
@@ -92,6 +96,65 @@ class Meta:
     __repr__ = sane_repr("id")
 
 
+@dataclass
+class SnowflakeBitSegment:
+    length: int
+    name: str
+
+    def validate(self, value):
+        if self.length <= 0:
+            raise Exception("The length should be a positive number")
+        if value >> self.length != 0:
+            raise Exception(f"{self.name} exceed max bit value of {self.length}")
+        return True
+
+
+class Snowflake:
```

Comment:
@maxiuyuan I think this doesn't need to be a class. Just exporting a function called `generate_snowflake_id()` should be sufficient.

Reply:
I agree that this doesn't need to be a class as is.

There are a lot of snowflake-specific constants, and I appreciate the effort to group them all as class variables. However, it would be even better just to create a `snowflake.py` file for all of this to live in. Then you could also remove the word "snowflake" from the beginning of all the names.

It might make sense to encapsulate the operation of creating one new snowflake ID into a class, if you find yourself repeatedly passing the same variables as arguments around multiple helper functions.

Reply:
Having a `snowflake.py` file can emulate the encapsulation that the `Snowflake` class provides.  👍 

Another motivation for making this a function rather than a class is that the snowflake instance will be short lived.

Also we can shorten the boilerplate code from this:

```python
        snowflake = Snowflake()
        if not self.id:
            self.id = snowflake.snowflake_id_generation()
```

to be just:

```python
import snowflake
# bunch of code...
        if not self.id:
            self.id = snowflake.generate_snowflake_id()
```